app/controllers/catalog_controller.rb

```diff
@@ -540,7 +541,7 @@ def st_upload_image
     elsif params[:upload] && params[:upload][:image] &&
           params[:upload][:image].respond_to?(:read)
       ext = params[:upload][:image].original_filename.split(".").last.downcase
```

Comment:
This could be moved into the else condition now cause it's not needed in the first condition anymore